packages/@expo/server/src/index.ts

```diff
@@ -0,0 +1,254 @@
+import '@expo/server/install';
+
+import { Response } from '@remix-run/node';
+import type { ExpoRoutesManifestV1, RouteInfo } from 'expo-router/build/routes-manifest';
+import fs from 'fs';
+import path from 'path';
+import { URL } from 'url';
+
+import { ExpoRequest, ExpoResponse, ExpoURL, NON_STANDARD_SYMBOL } from './environment';
+import { ExpoRouterServerManifestV1FunctionRoute } from './types';
+
+const debug = require('debug')('expo:server') as typeof console.log;
+
+function getProcessedManifest(path: string): ExpoRoutesManifestV1<RegExp> {
+  // TODO: JSON Schema for validation
+  const routesManifest = JSON.parse(fs.readFileSync(path, 'utf-8')) as ExpoRoutesManifestV1;
+
+  const parsed: ExpoRoutesManifestV1<RegExp> = {
+    ...routesManifest,
+    notFoundRoutes: routesManifest.notFoundRoutes.map((value: any) => {
+      return {
+        ...value,
+        namedRegex: new RegExp(value.namedRegex),
+      };
+    }),
+    apiRoutes: routesManifest.apiRoutes.map((value: any) => {
+      return {
+        ...value,
+        namedRegex: new RegExp(value.namedRegex),
+      };
+    }),
+    htmlRoutes: routesManifest.htmlRoutes.map((value: any) => {
+      return {
+        ...value,
+        namedRegex: new RegExp(value.namedRegex),
+      };
+    }),
+  };
+
+  return parsed;
+}
+
+export function getRoutesManifest(distFolder: string) {
+  return getProcessedManifest(path.join(distFolder, '_expo/routes.json'));
+}
+
+// TODO: Reuse this for dev as well
+export function createRequestHandler(
+  distFolder: string,
+  {
+    getRoutesManifest: getInternalRoutesManifest,
+    getHtml = async (request, route) => {
+      // serve a static file
+      const filePath = path.join(distFolder, route.page + '.html');
+
+      if (!fs.existsSync(filePath)) {
+        return null;
+      }
+      return fs.readFileSync(filePath, 'utf-8');
+    },
+    getApiRoute = async (route) => {
+      const filePath = path.join(distFolder, '_expo/functions', route.page + '.js');
+
+      debug(`Handling API route: ${route.page}: ${filePath}`);
+
+      // TODO: What's the standard behavior for malformed projects?
```

Comment:
🤷 Probably just throw an error in CLI, and return 500 as server error with a nice UI?

![image](https://github.com/expo/expo/assets/1203991/24b70f3c-32f6-4671-a135-d83d2402de25)

Reply:
looks good than, but we probably want something more unified in the future.